### PR TITLE
feat: add monitoring for Base and ZkSync inside l2-bridge-ethereum bot

### PR DESCRIPTION
**Pull request by sergeyWh1te**

### CODE REVIEW

File: l2-bridge-ethereum/src/constants.ts [hunk 1]
```diff
@@ -38,17 +38,34 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
 export const ARBITRUM_L1_GATEWAY_ROUTER =
   "0x72ce9c846789fdb6fc1f34ac4ad25dd9ef7031ef";
 
+export const ARBITRUM_L1ERC20_TOKEN_GATEWAY =
+  "0x0F25c1DC2a9922304f2eac71DCa9B07E310e8E5a";
+export const OPTIMISM_L1ERC20_TOKEN_BRIDGE =
+  "0x76943c0d61395d8f2edf9060e1533529cae05de6";
+export const BASE_L1ERC20_TOKEN_BRIDGE =
+  "0x9de443AdC5A411E83F1878Ef24C3F52C61571e72";
+export const ZKSYNC_L1ERC20_TOKEN_BRIDGE =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:47]:
```suggestion
export const ZKSYNC_L1ERC20_BRIDGE =
```

File: l2-bridge-ethereum/src/constants.ts [hunk 2]
```diff
@@ -38,17 +38,34 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
 export const ARBITRUM_L1_GATEWAY_ROUTER =
   "0x72ce9c846789fdb6fc1f34ac4ad25dd9ef7031ef";
 
+export const ARBITRUM_L1ERC20_TOKEN_GATEWAY =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:41]: Please consider adding proxy monitoring for `L1Executor` (zkSync part):

```js
export const ZKSYNC_L1EXECUTOR = 
    "0xFf7F4d05e3247374e86A3f7231A2Ed1CA63647F2";
```

File: l2-bridge-ethereum/src/constants.ts [hunk 3]
```diff
@@ -38,17 +38,34 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
...
+  "0x0F25c1DC2a9922304f2eac71DCa9B07E310e8E5a";
+export const OPTIMISM_L1ERC20_TOKEN_BRIDGE =
+  "0x76943c0d61395d8f2edf9060e1533529cae05de6";
+export const BASE_L1ERC20_TOKEN_BRIDGE =
+  "0x9de443AdC5A411E83F1878Ef24C3F52C61571e72";
+export const ZKSYNC_L1ERC20_TOKEN_BRIDGE =
+  "0x41527B2d03844dB6b0945f25702cB958b6d55989";
+
 export const GATEWAY_SET_EVENT =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:50]:
```suggestion
export const ARBITRUM_GATEWAY_SET_EVENT =
```

File: l2-bridge-ethereum/src/constants.ts [hunk 4]
```diff
@@ -38,17 +38,34 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
 export const ARBITRUM_L1_GATEWAY_ROUTER =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:38]: Please consider:

1) Using this constant below
image

2) Adding Base OVM L1 Cross Domain messenger here to monitor for the same administrative events:

> L1CrossDomainMessenger	[0x866E82a600A1414e583f7F13623F1aC5d58b0Afa](https://etherscan.io/address/0x866E82a600A1414e583f7F13623F1aC5d58b0Afa)
taken from here: https://docs.base.org/base-contracts#l1-contract-addresses
ABIs and events are the same as for Optimism

3) `DiamonProxy` events from zkSync

> DiamondProxy	[0x32400084c286cf3e17e7b677ea9583e60a000324](https://etherscan.io/address/0x32400084c286cf3e17e7b677ea9583e60a000324#code)

:warning:  Please pay attention to ABIs and events; they might be different for zkSync

**Review by TheDZhon — CHANGES_REQUESTED** [review]
Thank you for shipping the change.

However, I suggest polishing up the PR's content

File: l2-bridge-ethereum/src/constants.ts (hunk 2 above)
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/constants.ts:50]: added in another pr

File: l2-bridge-ethereum/src/abi/ProxyShortABI.json
[the remark this answers is not in the document]
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/abi/ProxyShortABI.json:1]: done

File: l2-bridge-ethereum/src/constants.ts (hunk 1 above)
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/constants.ts]: done

File: l2-bridge-ethereum/src/constants.ts (hunk 3 above)
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/constants.ts:60]: Do you know is the "event GatewaySet(address indexed l1Token, address indexed gateway)" same for optimism, base, arbitrum networks?

File: l2-bridge-ethereum/src/constants.ts (hunk 4 above)
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/constants.ts:38]: done

File: l2-bridge-ethereum/src/constants.ts [hunk 5]
```diff
@@ -38,17 +38,44 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
...
   "0x72ce9c846789fdb6fc1f34ac4ad25dd9ef7031ef";
 
-export const GATEWAY_SET_EVENT =
+export const OPTIMISM_L1_GATEWAY_ROUTER =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:41]: let's call it not 'GATEWAY_ROUTER` as it's a name for the Arbitrum's violet facade contract

it's `OPTIMISM_L1_CROSS_DOMAIN_MESSENGER` (L1CrossDomainMessenger here: https://community.optimism.io/docs/useful-tools/networks/#op-mainnet)

File: l2-bridge-ethereum/src/constants.ts [hunk 6]
```diff
@@ -38,17 +38,44 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
...
+  "0x25ace71c97b33cc4729cf772ae268934f7ab5fa1";
+
+export const BASE_L1_GATEWAY_ROUTER =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:44]:
```suggestion
export const OPTIMISM_L1_CROSS_DOMAIN_MESSENGER =
```

File: l2-bridge-ethereum/src/constants.ts [hunk 7]
```diff
@@ -38,17 +38,44 @@ export const WSTETH_ADDRESS = "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0";
...
+  "0x866E82a600A1414e583f7F13623F1aC5d58b0Afa";
+
+export const ZKSYNC_L1_DIAMOND_PROXY =
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:47]: :heart:

File: l2-bridge-ethereum/src/constants.ts [hunk 8]
```diff
@@ -192,6 +246,42 @@ export const THIRD_PARTY_PROXY_EVENTS = [
...
+      `was upgraded form: ${args._oldAddress} to: ${args._newAddress}`,
+    severity: FindingSeverity.High,
+    type: FindingType.Info,
+  },
+  {
+    address: ZKSYNC_L1_DIAMOND_PROXY, // ZkSync: DIAMOND proxy has changed
+    event:
+      "event DiamondCut(FacetCut[] facetCuts, address initAddress, bytes initCalldata)",
+    alertId: "THIRD-PARTY-PROXY-DIAMOND-CUT-CHANGED",
+    name: "🚨 ZkSync Native Bridge: OVM L1 Cross Domain Messenger proxy changed",
```
Comment (TheDZhon) [inline l2-bridge-ethereum/src/constants.ts:278]: :warning:  there is no 'OVM' for zkSync
please call it 'zkSync Diamond Proxy'

**Review by TheDZhon — CHANGES_REQUESTED** [review]
I belive we are almost done :pray:

File: l2-bridge-ethereum/src/constants.ts (hunk 6 above)
Comment (sergeyWh1te) [reply l2-bridge-ethereum/src/constants.ts]: I think you mean BASE_L1_CROSS_DOMAIN_MESSENGER
done

**Review by TheDZhon — APPROVED** [review]
Splendid :star_struck:

**Pull request merged**